src/crate/client/http.py

```diff
@@ -365,7 +365,8 @@ def blob_exists(self, table, digest):
     def _add_server(self, server):
         with self._lock:
             if server not in self.server_pool:
-                self.server_pool[server] = Server(server, **self._pool_kw)
+                kwargs = _remove_certs_for_non_https(server, self._pool_kw)
```

Comment:
This is the same procedure as in line 288-289.
You could extract these two lines into a new method and reuse the method.